src/angularWhitespaceRule.ts

```diff
@@ -152,11 +152,21 @@ class WhitespaceTemplateVisitor extends BasicTemplateAstVisitor {
 
 class PipeWhitespaceVisitor extends RecursiveAngularExpressionVisitor implements ConfigurableVisitor {
   visitPipe(ast: ast.BindingPipe, context: RecursiveAngularExpressionVisitor): any {
-    const start = ast.span.start;
-    const exprStart = context.getSourcePosition(ast.exp.span.start);
-    const exprEnd = context.getSourcePosition(ast.exp.span.end);
-    const sf = context.getSourceFile().getFullText();
-    const exprText = sf.substring(exprStart, exprEnd);
+
+    let exprStart, exprEnd, exprText, sf;
+    if (NgWalker.prop && ( NgWalker.prop.templateName === 'ngForOf' || NgWalker.prop.templateName === 'ngIf')) {
```

Comment:
Maybe we can preserve the context (`prop`) from a parent element visitor?

Reply:
Yes, this is an evil way. After a good night, now, we have our context the parent class `sourceMappingVistor`. It's better.

Reply:
Finally, I modified `SourceMappingVisitor.getSourcePosition()` method, which is more central. So, this fix the position problem in our tests.